### PR TITLE
WebGPU: Supports to IBL + PBR

### PR DESCRIPTION
**Pull request by sunag**
**Description**

This is the first implementation of IBL over PBR in WebGPU. 

This implementation open some issues related with `ShaderNode` that I will fix in the next days and weeks.

The implementation is not finished yet, but I think that is a good opportunity to merger as we will be working until `r141` release.

*Live*
https://raw.githack.com/sunag/three.js/dev-webgpu-pbr/examples/webgpu_loader_gltf.html

image

*This contribution is funded by [Google via Igalia](https://igalia.com)*

### CODE REVIEW

File: examples/webgpu_compute.html [hunk 1]
```diff
@@ -99,7 +99,7 @@
 					const pointer = uniform( pointerVector );
 					const limit = uniform( scaleVector );
 
-					const position = temp( context( add( particle, velocity ), { temp: false } ) );
+					const position = temp( add( particle, velocity ), 'tempPos' ); // @TODO: this should work without 'tempPos' property name
```
Comment (LeviPesin) [inline examples/webgpu_compute.html:102]: Then you can remove the import of `context`.

**Comment by LeviPesin** [comment]
I think we can now remove [these two lines](https://github.com/mrdoob/three.js/blob/c1094effe13475a2907bc8f4cd251b02e3e3c646/examples/jsm/renderers/webgpu/nodes/WebGPUNodeBuilder.js#L100-L101) from the WebGPUNodeBuilder?

Also, can you please add new nodes and functions to the ShaderNode?

**Comment by sunag** [comment]
> Also, can you please add new nodes and functions to the ShaderNode?

I don't think it's the best add all nodes in `ShaderNode`, because that loses in tree-shaking capabilities, for example: `Phong lighting` when we added this I think that is better to be outside the `ShaderNode`. So I hope to do it more strategically going forward.

**Comment by LeviPesin** [comment]
I think it is best to export from the ShaderNode the same nodes as exported from Nodes.js, and if some nodes are "internal" (i.e. not for external user user), we can remove them from both...

And what about functions?

**Comment by sunag** [comment]
Merging to sync new updates...

**Pull request merged**